### PR TITLE
ports Harder windows from TG

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
ports ||| https://github.com/tgstation/tgstation/pull/45609 , https://github.com/tgstation/tgstation/pull/46335 , https://github.com/tgstation/tgstation/pull/45890 , https://github.com/tgstation/tgstation/pull/46679 |||
tl;dr makes reinforced windows actually reinforced.

i do not take any credit for this i am only porting it to bee.

## Why It's Good For The Game
reinforced windows right now are pathetic, anyone can destroy them with a fire extinguisher in 20 hits, or simply deconstruct them with a screwdriver and crowbar, meanwhile plasma windows are very tough but you can still just get rid of them with a screwdriver and crowbar, these changes make reinforced windows way tougher and adds a lot more deconstruction steps to them.

## Changelog
:cl:
balance: rebalanced reinforced windows to be way tougher and harder to deconstruct.
/:cl:

### CODE REVIEW

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -115,7 +115,11 @@ Class Procs:
 	var/atom/movable/occupant = null
 	var/speed_process = FALSE // Process as fast as possible?
 	var/obj/item/circuitboard/circuit // Circuit to be created and inserted when the machinery is created
-	var/damage_deflection = 0
+	var/obj/item/card/id/inserted_scan_id
+	var/obj/item/card/id/inserted_modify_id
+	var/list/region_access = null // For the identification console (card.dm)
+	var/list/head_subordinates = null // For the identification console (card.dm)
```
Comment (zeskorion) [inline code/game/machinery/_machinery.dm]: What are theaw new vars in here for

**Comment by ghost** [comment]
well it was worth a try

**Comment by Isy232** [comment]
The issue with this change is too many windows on the station are reinforced.
Better solution is a map edit to make VERY important windows and external windows plasma glass.

**Comment by GoatMonarch** [comment]
From looking around Boxstation's web map the only areas you'd make more difficult to break into are:
Court Room, Brig cells (Electrified), Warden's officer (Electrified), every head office (electrified, within department), botany, EVA, snowflake tool storage, genetics (within department), Robotics&RnD(closeable shutters), every external window.

Every high security area most of the time are either surounded by renforced walls, or are in a high trafic area. Such as EVA being right beside bridge in the hallways. On the same path towards Security.

What diferentiates wherever high security areas are, are based uppon just electrifying the grilles. Insulateds are in low supply, especially round start. You'd get the same results by just making renforced windows drop glass shards with metal in them. This will ruin the meta of throwing glass on electrified grilles.
This will also lead to engineers taking much longer to make repairs. Or cause them to be able to spam an entire hallway with renforced glass using a rapid construction tool.

Though I fully agree with the construction paths for renforced plasma glass, as it's strenght is underwhelming.

**Comment by ghost** [comment]
> From looking around Boxstation's web map the only areas you'd make more difficult to break into are:
> Court Room, Brig cells (Electrified), Warden's officer (Electrified), every head office (electrified, within department), botany, EVA, snowflake tool storage, genetics (within department), Robotics&RnD(closeable shutters), every external window.
> 
> Every high security area most of the time are either surounded by renforced walls, or are in a high trafic area. Such as EVA being right beside bridge in the hallways. On the same path towards Security.
> 
> What diferentiates wherever high security areas are, are based uppon just electrifying the grilles. Insulateds are in low supply, especially round start. You'd get the same results by just making renforced windows drop glass shards with metal in them. This will ruin the meta of throwing glass on electrified grilles.
> This will also lead to engineers taking much longer to make repairs. Or cause them to be able to spam an entire hallway with renforced glass using a rapid construction tool.
> 
> Though I fully agree with the construction paths for renforced plasma glass, as it's strenght is underwhelming.

its been working perfectly on tg since over half a year now

File: code/game/machinery/_machinery.dm (hunk 1 above)
Comment (ghost) [reply code/game/machinery/_machinery.dm]: > 
> 
> What are theaw new vars in here for

i have no idea

**Comment by ghost** [comment]
i'm gonna nuke this branch and redo it without all the weird file changes aswell as some changes to windows.

**Comment by GoatMonarch** [comment]
> > From looking around Boxstation's web map the only areas you'd make more difficult to break into are:
> > Court Room, Brig cells (Electrified), Warden's officer (Electrified), every head office (electrified, within department), botany, EVA, snowflake tool storage, genetics (within department), Robotics&RnD(closeable shutters), every external window.
> > Every high security area most of the time are either surounded by renforced walls, or are in a high trafic area. Such as EVA being right beside bridge in the hallways. On the same path towards Security.
> > What diferentiates wherever high security areas are, are based uppon just electrifying the grilles. Insulateds are in low supply, especially round start. You'd get the same results by just making renforced windows drop glass shards with metal in them. This will ruin the meta of throwing glass on electrified grilles.
> > This will also lead to engineers taking much longer to make repairs. Or cause them to be able to spam an entire hallway with renforced glass using a rapid construction tool.
> > Though I fully agree with the construction paths for renforced plasma glass, as it's strenght is underwhelming.
> 
> its been working perfectly on tg since over half a year now

Cobbychem, removal of cloning, botany nerfs have worked as well for some time.
_Does not mean it's good changes though._

**Comment by knacker48** [comment]
As someone who used to play on TG before and after the change I have to say that it wasn't very good one. The extra steps for deconstruction just switched people from actually dismantling windows stealthy to just breaking them down since it was both less time consuming and you didn't need tools to do it

**Pull request closed without merging**